packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-config.ts

```diff
@@ -1,46 +1,128 @@
+import { ArnFormat, Resource, Stack } from '@aws-cdk/core';
 import { Construct } from 'constructs';
-import { arnForDeploymentConfig } from '../utils';
+import { CfnDeploymentConfig } from '../codedeploy.generated';
+import { TrafficRouting } from '../traffic-routing-config';
+import { arnForDeploymentConfig, validateName } from '../utils';
 
 /**
  * The Deployment Configuration of an ECS Deployment Group.
+ *
+ * If you're managing the Deployment Configuration alongside the rest of your CDK resources,
+ * use the {@link EcsDeploymentConfig} class.
+ *
+ * If you want to reference an already existing deployment configuration,
+ * or one defined in a different CDK Stack,
+ * use the {@link EcsDeploymentConfig#fromEcsDeploymentConfigName} method.
+ *
  * The default, pre-defined Configurations are available as constants on the {@link EcsDeploymentConfig} class
  * (for example, `EcsDeploymentConfig.AllAtOnce`).
- *
- * Note: CloudFormation does not currently support creating custom ECS configs outside
- * of using a custom resource. You can import custom deployment config created outside the
- * CDK or via a custom resource with {@link EcsDeploymentConfig#fromEcsDeploymentConfigName}.
  */
 export interface IEcsDeploymentConfig {
+  /**
+   * The physical, human-readable name of the Deployment Configuration.
+   * @attribute
+   */
   readonly deploymentConfigName: string;
+
+  /**
+   * The ARN of the Deployment Configuration.
+   * @attribute
+   */
   readonly deploymentConfigArn: string;
 }
 
+/**
+ * Construction properties of {@link EcsDeploymentConfig}.
+ */
+export interface EcsDeploymentConfigProps {
+  /**
+   * The physical, human-readable name of the Deployment Configuration.
+   * @default - automatically generated name
+   */
+  readonly deploymentConfigName?: string;
+
+  /**
+   * The configuration that specifies how traffic is shifted from the 'blue'
+   * target group to the 'green' target group during a deployment.
+   * @default AllAtOnce
+   */
+  readonly trafficRouting?: TrafficRouting;
+}
+
 /**
  * A custom Deployment Configuration for an ECS Deployment Group.
  *
- * Note: This class currently stands as namespaced container of the default configurations
- * until CloudFormation supports custom ECS Deployment Configs. Until then it is closed
- * (private constructor) and does not extend {@link Construct}
- *
  * @resource AWS::CodeDeploy::DeploymentConfig
  */
-export class EcsDeploymentConfig {
+export class EcsDeploymentConfig extends Resource implements IEcsDeploymentConfig {
```

Comment:
I think it might also make sense to create a command abstract base class `DeploymentConfigBase`. It looks like almost all of the logic between ECS and Lambda are the same.

Reply:
Done!